src/app/Dashboard/components/DashboardContainer.js

```diff
@@ -94,6 +109,69 @@ export class DashboardContainer extends Component {
     });
   };
 
+  handleClick = index => (event) => {
+    if (event.target.value === 'Select All') {
+      const filterBy = this.state.filterBy.map((item) => {
+        if (event.target.checked) {
+          return (item.checked = true), item;
+        }
+
+        return (item.checked = false), item;
+      });
+
+      this.setState({ filterBy, filteredUserActivities: null });
+    } else {
+      const filterBy = this.state.filterBy.slice();
+      filterBy[index].checked = !filterBy[index].checked;
+
+      if (!filterBy[index].checked) {
+        const filteredUserActivities = (
+          this.state.filteredUserActivities || this.props.userActivities
+        ).filter(activity => activity.status !== event.target.value);
+
+        if (!filteredUserActivities.length) {
+          this.setState({ filterBy, filteredUserActivities: null });
+          return;
+        }
+
+        this.setState({ filterBy, filteredUserActivities });
+        return;
+      }
+
+      if (this.state.filteredUserActivities) {
+        const filteredUserActivities = [
+          ...this.state.filteredUserActivities,
+          ...this.props.userActivities.filter(
+            activity => activity.status === event.target.value,
+          ),
+        ];
+
+        this.setState({ filterBy, filteredUserActivities });
+        return;
+      }
+
+      const filteredUserActivities = this.props.userActivities.filter(
+        activity => activity.status === event.target.value,
+      );
+      this.setState({ filterBy, filteredUserActivities });
+    }
+  };
+
+  hideFilter = (e) => {
+    if (!e.target.className) {
```

Comment:
I don't really think this is necessary, we can just have:

```
if (!this.filterRef.current.contains(e.target)) {
    this.setState({ show: false });
 }
```

Or any reason why we are negating the className in the target object?

Reply:
@ajimae Yes, if we remove that when you click the filter button the `showFilter` and `hideFilter`(`document.addEventListener('mousedown', this.hideFilter, false);`) that's why I check if the button has been click I don't do anything on `hideFilter`. otherwise the filter option will  disappear and show again